src/syscall.c

```diff
@@ -376,4 +380,12 @@ void syscall_handler(riscv_t *rv)
         fprintf(stderr, "unknown syscall %d\n", (int) syscall);
         break;
     }
+
+    /*
+     * save return code
+     *
+     * the application decides the usage of the return code
+     */
+    vm_attr_t *vm_attr = rv_userdata(rv); /* access userdata */
```

Comment:
The line `vm_attr_t *vm_attr = rv_userdata(rv);` might be perplexing due to the similar naming of the object type and the `vm_attr`, while the value comes from the `rv_userdata` call. This suggests a presumed direct link between `rv_userdata` and `vm_attr_t`. However, this connection is not explicitly clarified in the API documentation, leading to potential confusion regarding their relationship.

Reply:
Added more comments about them. Hopefully it is clearer now.